src/mon/MDSMonitor.cc

```diff
@@ -347,11 +350,11 @@ bool MDSMonitor::preprocess_beacon(MonOpRequestRef op)
   }
 
   // fw to leader?
-  if (!mon->is_leader())
+  if (!is_leader())
     return false;
 
   // booted, but not in map?
-  if (!pending_fsmap.gid_exists(gid)) {
+  if (!fsmap.gid_exists(gid)) {
```

Comment:
Correct me if I'm wrong, but this block is allowing sending a message to an MDS based on yet to be committed state, right? Is the MDS able to cope with this?

Reply:
I thought that was suspicious too but kept the existing behavior. I don't know.